Content.Client/UserInterface/Controls/EntityListDisplay.cs

```diff
@@ -1,292 +0,0 @@
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using Content.Client.Stylesheets;
-using JetBrains.Annotations;
-using Robust.Client.UserInterface;
-using Robust.Client.UserInterface.Controls;
-using Robust.Shared.GameObjects;
-using Robust.Shared.Maths;
-
-namespace Content.Client.UserInterface.Controls
-{
-    public class EntityListDisplay : Control
```

Comment:
EntityListDisplay was created to replace ItemList for the Entity Spawn panel, Cargo Console, and to a lesser extent Storage. The difference between using EntityListDisplay and creating a button for each entity is that EntityListDisplay only creates buttons for the visible items (usually 6-20) to reduce lag when opening a large list like for the Entity Spawn panel.

I sorta lost motivation for finishing it so it still needs work, but it does handle large lists better than creating every button. It needs some tweaking for removing only the buttons that are scrolled out of view and some changes for ease of use. Unless someone wants to help finish it, then I'm fine with it being removed, but when I have time I'll probably fix it up and readd it.

Reply:
Instead of just removing it I could also just leave it in or try fix it (at least so that #5180 is fixed), or replace it somehow.

Is the main issue actually the generation of buttons, or the generation of the button contents (i.e., getting entity/prototype sprites & names, and creating those controls). Because creating a modified `ScrollContainer` that has empty buttons of a known size and just fills in their contents as they come into view is relatively straightforward. But for the spawn-menu, I imagine that it might still be a problem when it has to create many blank buttons.

Reply:
The main problem is button+children. 

The Entity Spawn panel doesn't lag right now because PJB implemented the only show what's visible thing in the panel.

The EntityListDisplay was an attempt to generalize it more, but it looks like it needs more tweaks to the limits (That'd be under that big code comment). I just didn't think it was ready to replace the Entity Spawn panel one yet.

Reply:
For now, I'd want to just leave this PR as is just for the XAML & Icon, and then a open separate pair of PRs that adds some sort of control to the engine and makes both the storage UI and spawn-menu use it.

I started playing around with a control that just inherits from `ScrollContainer`. It means that there's less code duplication but it probably won't be as efficient as the current `EntityListDisplay`, and I haven't tested it with the spawn menu yet, so it might not work out and I'd just use `EntityListDisplay` with fixes for the disappearing entities. My attempt just creates buttons with some known (but individually variable) size and dynamically fills their contents in. This is a video of a with an **intentionally incorrect** definition of what it means for a button to be 'in-view' for demonstration purposes.

https://user-images.githubusercontent.com/60421075/143597609-7c2337d6-a2d3-42d8-8f06-9e30874ece40.mp4

Reply:
Here's my previous unmerged attempt with comments. Might help with search terms. https://github.com/space-wizards/RobustToolbox/pull/1166